llvm/lib/Target/RISCV/RISCVInstrInfoXCheri.td

```diff
@@ -872,6 +872,7 @@ def : PatGpcr<int_cheri_cap_seal_entry, CSealEntry>;
 
 // TODO: Make this rematerialisable like MIPS
 def : PatGpcrGpr<int_cheri_bounded_stack_cap, CSetBounds>;
+def : PatGpcrUimm12<int_cheri_bounded_stack_cap, CSetBoundsImm>;
```

Comment:
This intrinsic is an awful hack, but I couldn't come up with a reasonable way to get the MIPS tablegen patterns to work :(